### ISSUE TITLE
Hooks?

### ISSUE THREAD

**Issue by elfeffe** [artkonekt/concord#10]
Does it supports Hooks?
I have found this package https://github.com/tormjens/eventy
But does your package offers that functionality?

**Comment by fulopattila122** [comment]
No, apps written with Concord are typically using the default [Laravel Event system](https://laravel.com/docs/7.x/events) to achieve such functionality. Concord itself fires [a couple of events](https://github.com/artkonekt/concord/tree/1.6/src/Events) and also supports apps/modules to [disable events](https://konekt.dev/concord/1.6/event-listener-bindings).

Nevertheless, you can use the mentioned package along with Concord without any problems.

Does this properly answer your question?

**Comment by elfeffe** [comment]
Sorry, but can you show me how to do that? Because the conventional approach is very criticized because “listeners but not return values”. 
What I need is to create some event getPaymentMethods and the plugins must return their methods. Do you have any example?

**Comment by fulopattila122** [comment]
As an example, the Vanilo Payment module (which is built on top of Concord) is using the registry pattern to achieve such functionality: https://github.com/vanilophp/payment/blob/master/src/PaymentGateways.php#L22

And any plugin for the payment module can register itself with `PaymentGateways::register(...)`
Example plugin registering itself: https://github.com/artkonekt/vanilo-euplatesc/blob/master/src/Providers/ModuleServiceProvider.php#L26

**Comment by elfeffe** [comment]
Great, thanks a lot

**Comment by elfeffe** [comment]
@fulopattila122 I didn't see you are the developer, I have been following Vanilo for some time, from the first beta.
When do you plan to release the payments ans shipping modules? I think I can use them in my project.
